### PR TITLE
Updated to scriptcs 0.16.1 

### PR DESCRIPTION
**Pull request by filipw**
- removes dependency on old Roslyn compilers
 - support for C# 6

### CODE REVIEW

File: src/ConfigR/Scripting/ScriptConfigLoader.cs [hunk 1]
```diff
@@ -64,25 +64,25 @@ private static void RethrowExceptionIfAny(ScriptResult result, string scriptPath
         {
             if (result.CompileExceptionInfo != null)
             {
-                result.CompileExceptionInfo.Throw();
-            }
-
-            if (result.ExecuteExceptionInfo != null)
-            {
                 // HACK: waiting on https://github.com/scriptcs/scriptcs/issues/545
```
Comment (filipw) [inline src/ConfigR/Scripting/ScriptConfigLoader.cs]: this is changed because Roslyn 1.0.0-rc2 doesn't throw execution exception anymore for empty files

**Comment by adamralph** [comment]
Thanks @filipw - will review ASAP.

File: src/ConfigR/Scripting/ScriptConfigLoader.cs (hunk 1 above)
Comment (adamralph) [inline src/ConfigR/Scripting/ScriptConfigLoader.cs]: Does that mean https://github.com/scriptcs/scriptcs/issues/545 is fixed in 0.16.1?

Comment (adamralph) [inline src/ConfigR/Scripting/ScriptConfigLoader.cs]: I believe it is.

The problem now is that `CSharpScriptInMemoryEngine` still throws a compilation exception in the case of a no-op file, which means that, in 0.16.1, `scriptcs empty.csx` works, but `scriptcs empty.csx -debug` doesn't.

**Pull request merged**